From 1139ac302c5ac3e91e23fab9f03c98aac3951125 Mon Sep 17 00:00:00 2001
From: Angela Czubak <aczubak@marvell.com>
Date: Thu, 30 May 2019 16:08:33 +0200
Subject: [PATCH 287/386] octeontx: fix __get_sso_group_pend

The bitmask traversal in the function mentioned above is incomplete.
Macro for_each_set_bit() expect bitmask size to be given in bits and
not in bytes.
Previously only first 8 bits, and therefore SSOs, were drained. This
could lead to problems with restarting an application using SSOs, which
had VF IDs greater or equal 8, as they might have received invalid
pointers (especially not mapped into their virtual address space).

Change-Id: Ie70576e508dc5c70b3250bd4fbce9319a3a5316f
Signed-off-by: Angela Czubak <aczubak@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10276
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/ssowpf_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/ssowpf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/ssowpf_main.c
index 5c3edc0ae64b..426998c4ba10 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/ssowpf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/ssowpf_main.c
@@ -339,7 +339,7 @@ static int __get_sso_group_pend(u32 id, u64 grp_mask)
 	u64 reg;
 
 	for_each_set_bit(i, (const unsigned long *)&grp_mask,
-			 sizeof(grp_mask)) {
+			 sizeof(grp_mask) * 8) {
 		aq_cnt = 0;
 		cq_cnt = 0;
 		ds_cnt = 0;
-- 
2.17.1

